### ISSUE TITLE
[HunyuanVideo-1.5][QB2][frames=121] DiT OOMs DRAM on the 51275×51275 attention mask: unfused SDPA scores at opt level 0, fp32 mask copies at opt level 1

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-xla#6071]
### Problem description

With the ttnn.pad L1 fix (tenstorrent/tt-metal#56121) applied, the HunyuanVideo-1.5 DiT at `frames=121` (S = 49290 video + 1985 text = 51275 tokens, QB2, 1x4 mesh) gets past program creation but OOMs device DRAM in the joint attention. Three variants were hit:

1. **`optimization_level=0` – unfused SDPA.** The bool mask skips the SDPA composite, attention is decomposed, and the `(1, 4, 51275, 51275)` f32 scores tensor (42.1 GB) is materialized:
   ```
   TT_FATAL: Out of Memory: Not enough space to allocate 42100473856 B DRAM buffer across 8 banks
   ```

2. **`optimization_level=1` – fused SDPA, leftover f32 mask.** The optimizer fuses the decomposition back into `ttnn.scaled_dot_product_attention` (bf16 mask), but the native path's fully-masked-row handling (`max`/`eq`/`where` over the mask) survives and needs an f32 copy of the `(1, 1, 51275, 51275)` mask (10.5 GB) in every block. Blocks 0–29 ran; block 30 OOMed:
   ```
   TT_FATAL: Out of Memory: Not enough space to allocate 10525118464 B DRAM buffer across 8 banks,
   where each bank needs to store 1315643392 B ... largest free block: 1310821440 B
   ```
   Block 30's graph is identical to the 30 before it (682 TTIR lines, same op list). What changed is memory: the per-bank baseline grows by a constant 37.3 MB per block as each block's weights land on device, while each block peaks ~2.15 GB above its baseline. At block 30 the f32 mask needed 1315.6 MB contiguous per bank and the largest free block was 1310.8 MB.

   | Block | Baseline / bank | Peak / bank | Largest free block at start |
   |---|---|---|---|
   | 0 | 183 MB | 2334 MB | 3837 MB |
   | 10 | 554 MB | 2632 MB | 3458 MB |
   | 20 | 927 MB | 3005 MB | 3121 MB |
   | 29 | 1337 MB | 3414 MB | 2699 MB |
   | 30 | 1374 MB | OOM | 2673 MB |

3. **`optimization_level=1` + `TT_XLA_CATCH_BOOL_MASK_SDPA=1` – composite mask conversion in f32.** With the flag, the SDPA composite captures the bool mask, removing the leftover chain above. But its bool→additive conversion `torch.where(mask, 0.0, -inf).to(query.dtype)` resolves to f32 (Python float scalars), so two f32 copies of the mask (21 GB) are live before the bf16 result exists. Peak rose to ~2.68 GB/bank above baseline and the OOM moved earlier, to block 14.

### Fix

- [PR](https://github.com/tenstorrent/tt-xla/pull/6072) – build the `where` branches in `query.dtype` in `composite_scaled_dot_product_attention` so the additive mask is produced directly in bf16, with no f32 round trip. Per-block peak drops to ~1.35 GB/bank (two bf16 mask copies), and all 54 blocks fit.
- Test config: `optimization_level=1` and `TT_XLA_CATCH_BOOL_MASK_SDPA=1` on the transformer component test.

| | |
|---|---|
| Result | PASSED, PCC 0.980469 (threshold 0.90) |
| TT phase | 29 min 39 s, all 54 blocks |
| CPU golden | 32 min |
| Total | 1 h 1 min 55 s |

### Next steps

- Once the tt-metal pad fix reaches tt-xla via the tt-mlir uplift and this [PR](https://github.com/tenstorrent/tt-xla/pull/6072) lands, run the e2e pipeline so the transformer sees its real inputs.
- If PCC holds, add `num_frames=121` support with `num_inference_steps=10` in the PCC-gated test and 50 in the demo and benchmark. The step count for the PCC test is bounded by the CPU golden: one DiT forward on CPU took ~32 min in this run, and the real pipeline inputs should cost about the same, so 50 steps would put the CPU golden alone past a day, while 10 steps keeps it around 5 h and it can be cut to 5 steps if that is still too long for CI.

### Repro

```
cd third_party/tt_forge_models
git checkout kkannan/sep10_hv1_5_num_frames_121
cd ../..

# add the attached scripts and test it
```

### Logs

- [failures_and_fixes.zip](https://github.com/user-attachments/files/32143753/next_failure.zip)
